### ISSUE TITLE
Vert.x - No ReactiveStreamsFactory implementation found!

### ISSUE THREAD

**Issue by p-zalejko** [quarkusio/quarkus#3605]
I started playing with Reactive Streams and ran into a problem with a missing implementation of the `ReactiveStreamsFactory`.

**To Reproduce**
Follow this tutorial https://quarkus.io/guides/using-vertx :
1. Create a hello world application
2. Add `vertx` extension
3. Implement a simple endpoint (using the Axle provider)
4. Build a native app

My pom.xml dependencies looked as follows:

```
<dependency>
  <groupId>io.quarkus</groupId>
  <artifactId>quarkus-resteasy</artifactId>
</dependency>
<dependency>
  <groupId>org.jetbrains.kotlin</groupId>
  <artifactId>kotlin-stdlib-jdk8</artifactId>
</dependency>
<dependency>
  <groupId>io.quarkus</groupId>
  <artifactId>quarkus-junit5</artifactId>
<scope>test</scope>
</dependency>
<dependency>
  <groupId>io.rest-assured</groupId>
  <artifactId>rest-assured</artifactId>
  <scope>test</scope>
</dependency>
<dependency>
  <groupId>io.quarkus</groupId>
  <artifactId>quarkus-vertx</artifactId>
</dependency>
```
The endpoint as this:
```

import io.vertx.axle.core.Vertx
import org.reactivestreams.Publisher
import javax.inject.Inject
import javax.ws.rs.GET
import javax.ws.rs.Path;
import javax.ws.rs.Produces
import javax.ws.rs.core.MediaType


@Path("/events")
class EventController {

  @Inject
  lateinit var vertx: Vertx

  @GET
  @Produces(MediaType.SERVER_SENT_EVENTS)
  fun getAll(): Publisher<String> {
    val toPublisherBuilder = vertx.periodicStream(200).toPublisherBuilder()
    return toPublisherBuilder
           .map { "hello" }
           .buildRs()
  }
}
```
**Issue**
After launching the application I got the following exception:
```
Caused by: java.lang.IllegalStateException: No ReactiveStreamsFactory implementation found!
at org.eclipse.microprofile.reactive.streams.operators.spi.ReactiveStreamsFactoryResolver.instance(ReactiveStreamsFactoryResolver.java:64)
at org.eclipse.microprofile.reactive.streams.operators.ReactiveStreams.fromPublisher(ReactiveStreams.java:60)
at io.vertx.axle.core.TimeoutStream.toPublisherBuilder(TimeoutStream.java:93)
...
```
Turned out that the `ReactiveStreamsFactory` implementation is missing. I found that it can be fixed by adding `"io.quarkus:quarkus-smallrye-reactive-streams-operators"` to the pom.xml:
```
<dependency>
  <groupId>io.quarkus</groupId>
  <artifactId>quarkus-smallrye-reactive-streams-operators</artifactId>
</dependency>
```
**Solution**
A similar issue has been already reported here https://github.com/quarkusio/quarkus/issues/1495 . I was fixed by adding it as a dependency.

How about adding `quarkus-smallrye-reactive-streams-operators` as a dependency also to vert.x extension?

**Comment by gsmet** [comment]
@cescoffier can you take care of this one?

**Comment by p-zalejko** [comment]
I created a PR with the same change as its was made for https://github.com/quarkusio/quarkus/issues/1495 (I added this dependency to the deployment module). Unfortunately I cannot verify if it solves the problem. My OS is Fedora 30 and I'm not able to build native images due to https://github.com/quarkusio/quarkus/issues/3526

**Comment by cescoffier** [comment]
Fixed by #3611

**Comment by gsmet** [comment]
Reopening as we had to revert it in https://github.com/quarkusio/quarkus/pull/3896 .

The current fix causes issues as we need the `-deployment` dependency to be in the deployment artifact and the `-runtime` dependency to be in the runtime artifact.

The proposed fix mistakenly add the runtime dependency to the deployment artifact.

That being said, I tried to do it properly and we have a circular dependency due to some test dependencies creating a circular dependency. We will probably need to move these tests to some IT.
